### PR TITLE
vi bid adapter

### PR DESCRIPTION
**Pull request by SnoopInf**
## Type of change
- [ X ] New bidder adapter   

## Description of change
vi bid adapter

- test parameters for validating bids
```
{
   bidder: 'vi',
   params: {
            pubId: 'sb_test',
            lang: 'en-US',
            cat: 'IAB1',
	        bidFloor: 0.05 //optional
  }
}
```

- contact email of the adapter’s maintainer
   support@vi.ai

- [ X ] official adapter submission

## Other information
https://github.com/prebid/prebid.github.io/pull/537

### CODE REVIEW

File: test/spec/modules/viBidAdapter_spec.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { expect } from 'chai';
+import { spec } from 'modules/viBidAdapter';
+import { newBidder } from 'src/adapters/bidderFactory';
+import { REPO_AND_VERSION } from 'src/constants';
```
Comment (matthewlane) [inline test/spec/modules/viBidAdapter_spec.js]: This import isn't used in the test, line can be dropped

File: modules/viBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+function interpretResponse(bids) {
+  let responses = [];
+  utils._each(bids.body, function(bid) {
+    responses.push({
+      requestId: bid.id,
+      cpm: parseFloat(bid.price),
+      width: parseInt(bid.width),
+      height: parseInt(bid.height),
```
Comment (matthewlane) [inline modules/viBidAdapter.js]: A radix (likely, `10`) should be specified as the second argument to the above two `parseInt`s to guarantee predictable behavior

**Comment by matthewlane** [comment]
Docs PR for this adapter at https://github.com/prebid/prebid.github.io/pull/537

**Comment by SnoopInf** [comment]
Removed import and added radix

File: modules/viBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+function isBidRequestValid(bid) {
+  return !!(bid.params.pubId);
+}
+
+function buildRequests(bidReqs) {
+  let imps = [];
+  utils._each(bidReqs, function (bid) {
+    imps.push({
+      id: bid.bidId,
+      sizes: bid.sizes,
```
Comment (matthewlane) [inline modules/viBidAdapter.js]: ad unit sizes may be defined as a double array (`[[320, 480]]`) or single array (`[320, 480]`). In the case of single sizes, your endpoint is erroring out with `400 (Invalid JSON: Cannot deserialize instance of int[] out of VALUE_NUMBER_INT`, but `utils.parseSizesInput` can help. Something like
```
utils.parseSizesInput(bid.sizes).map(size => size.split('x'))
```
though feel free handle it however you'd like

**Comment by ndhimehta** [comment]
Assigning to @snapwich for a second round of review.

**Comment by SnoopInf** [comment]
Thank you guys!
What's our next action? how to make plugin listed on Prebid.js web site?

**Comment by matthewlane** [comment]
@SnoopInf This adapter will be in the next Prebid release, which should be happening this week. When that happens we'll merge the docs PR and vi will be listed on the site. Thanks for contributing to the project 👍

**Pull request merged**